core/dbt/parser/base.py

```diff
@@ -71,20 +71,44 @@ def __init__(
 
 class RelationUpdate:
     def __init__(self, config: RuntimeConfig, manifest: Manifest, component: str) -> None:
-        macro = manifest.find_generate_macro_by_name(
+        default_macro = manifest.find_generate_macro_by_name(
             component=component,
             root_project_name=config.project_name,
         )
-        if macro is None:
+        if default_macro is None:
             raise DbtInternalError(f"No macro with name generate_{component}_name found")
 
-        root_context = generate_generate_name_macro_context(macro, config, manifest)
-        self.updater = MacroGenerator(macro, root_context)
+        default_macro_context = generate_generate_name_macro_context(
+            default_macro, config, manifest
+        )
+        self.default_updater = MacroGenerator(default_macro, default_macro_context)
+
+        package_names = {macro.package_name for macro in manifest.macros.values()}
```

Comment:
@gshank - this doesn't seem too bad to do for each RelationUpdate instantiation (3 - one per component) per manifest (1-2). Is there somewhere where this information is already loaded during parsing? We could add it to the manifest but that would increase the scope of this PRs changes pretty significantly so I'd opt to do it as a follow-on if that feels like the right way forward.

Reply:
I updated this bit to grab the dependency names from RuntimeConfig instead of computing them based on nodes in the manifest 👍